lesson18/src/js/main.js

```diff
@@ -0,0 +1,299 @@
+function createElementWithClassName(element, name) {
+  const createElement = document.createElement(element);
+  createElement.classList.add(name);
+  return createElement;
+}
+
+function showLoadingImage() {
+  const main = document.getElementById("main");
+  const renderLoadingBox = createElementWithClassName("div", "loading");
+  renderLoadingBox.id = "js-loading";
+  const loadingImage = createElementWithClassName("img", "loading__image");
+  loadingImage.src = "images/loading.gif";
+  loadingImage.width = 80;
+  loadingImage.height = 80;
+  renderLoadingBox.appendChild(loadingImage);
+  document.body.insertBefore(renderLoadingBox, main);
+}
+
+function removeLoading() {
+  document.getElementById("js-loading").remove();
+}
+
+function displayErrorMassage(error) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+  const textBox = createElementWithClassName("p", "error-message");
+  textBox.textContent = error;
+  slideImageArea.appendChild(textBox);
+}
+
+function renderSlideArea() {
+  const slideImageArea = createElementWithClassName(
+    "div",
+    "mainvisual__images__inner"
+  );
+  document.querySelector(".mainvisual__images").appendChild(slideImageArea);
+}
+
+function renderSlideImage(item) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+  const slider = createElementWithClassName("ul", "slider");
+  const fragment = document.createDocumentFragment();
+  item.forEach((image, i) => {
+    const sliderItem = createElementWithClassName("li", "slider__item");
+    sliderItem.dataset.slideIndex = i + 1;
+
+    const sliderImage = createElementWithClassName("img", "slider__image");
+    sliderImage.src = image.image;
+    sliderImage.width = image.width;
+    sliderImage.height = image.height;
+
+    i === 0 && sliderItem.classList.add("is-active");
+
+    fragment.appendChild(sliderItem).appendChild(sliderImage);
+  });
+
+  slideImageArea.appendChild(slider).appendChild(fragment);
+}
+
+function renderSlideButton() {
+  let fragment = document.createDocumentFragment();
```

Comment:
IMO
I think `const` is fine, not `let`.

Reply:
The constant is appropriate!
I fixed this.

[b6f60e6](https://github.com/nyattsu72/hanson-frontend-cource/pull/26/commits/b6f60e63316e53ed1da79e2f3d7dcf7ce78d2c22)